feature-libs/product/configurators/common/src/components/config-previous-next-buttons/config-previous-next-buttons.component.ts

```diff
@@ -51,12 +58,17 @@ export class ConfigPreviousNextButtonsComponent {
   }
 
   protected scrollToVariantConfigurationHeader(): void {
-    const theElement = document.querySelector('.VariantConfigurationTemplate');
-    let topOffset = 0;
-    if (theElement instanceof HTMLElement) {
-      topOffset = theElement.offsetTop;
+    if (isPlatformBrowser(this.platformId)) {
+      // we don't want to run this logic when doing SSR
+      const theElement = document.querySelector(
```

Comment:
Injecting the platform is a way to go 👍 
But maybe we can actually inject `protected windowRef: WindowRef` instead, and try the following?

```ts
  protected scrollToVariantConfigurationHeader(): void {
    const theElement = this.renderer.selectRootElement(
      '.VariantConfigurationTemplate'
    );
    let topOffset = 0;
    if (theElement instanceof HTMLElement) {
      topOffset = theElement.offsetTop;
    }
    this.windowRef.nativeWindow?.scroll(0, topOffset);
  }
```

Reply:
Actually, scratch the above.
Let's extract the logic as is, into a service which can be re-used by other components.
Please make the class name in the service override-able. And the service should be smart enough to know whether to actually do the scrolling, based on the platform used.

Reply:
Just for clarificaton: should I extract the logic as is meaning as it was before or with the usage of the renderer?

Reply:
Yes, before the renderer, and wrapped around the platform check

Reply:
I have moved it to config-utils.service.ts as "scrollToConfigurationElement(selectors: string): void ". 
Currently config-utils.service.ts is located in storefrontlib but will be moved to the lib later. I've added a todo for this in todo.md.